File: typescript/resistor-color/resistor-color.ts
export const colorCode = (color: string) => {
  let index = COLORS.indexOf(color)
  if (index === -1) {
    throw new Error("error color value")
  }
  return index
}

export const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white',
]

File: java/resistor-color/src/main/java/ResistorColor.java
import java.util.HashMap;
import java.util.Map;

class ResistorColor {
    private final Map<String, Integer> COLOR_CODES = new HashMap<>() {
        {
            put("black", 0);
            put("brown", 1);
            put("red", 2);
            put("orange", 3);
            put("yellow", 4);
            put("green", 5);
            put("blue", 6);
            put("violet", 7);
            put("grey", 8);
            put("white", 9);
        }
    };

    int colorCode(String color) {
        return COLOR_CODES.get(color);
    }

    String[] colors() {
        return COLOR_CODES.entrySet().stream()
                .sorted(Map.Entry.comparingByValue())
                .map(Map.Entry::getKey)
                .toArray(String[]::new);
    }
}
